File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(auto a : commands){
        int start = a[0];
        int end = a[1];
        int pivot = a[2];
        vector<int> v;
        
        for(int i = start-1; i < end; i++){
            v.push_back(array[i]);
        }
        sort(v.begin(), v.end());
        
        answer.push_back(v[pivot - 1]);
        
    }
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        int idx = 0;
        
        for(int[] command : commands){
            int i = command[0];
            int j = command[1];
            int k = command[2];
            
            List<Integer> temp = new ArrayList<>();
            for(int t = i - 1; t < j; t++)
                temp.add(array[t]);
            
            Collections.sort(temp);
            answer[idx] = temp.get(k - 1);
            idx++; 
        }
        
        return answer;
    }
}
